lib/api/controllers/adminController.js

```diff
@@ -46,49 +41,24 @@ class AdminController extends BaseController {
       'loadSecurities',
       'resetCache',
       'resetDatabase',
-      'resetKuzzleData',
       'resetSecurity',
       'shutdown'
     ]);
   }
 
-
-  /**
-   * Reset the internal storage components (internal index, cache and public storage)
-   */
-  resetKuzzleData (request) {
-    this._lockAction(request, 'Kuzzle is already resetting his data.');
-
-    return this.kuzzle.repositories.user.truncate({ refresh: 'wait_for' })
-      .then(() => this.kuzzle.internalIndex.deleteIndex())
-      .then(() => this.kuzzle.services.internalCache.flushdb())
-      .then(() => this.kuzzle.internalIndex.bootstrap.startOrWait())
-      .then(() => {
-        delete _locks.resetKuzzleData;
-
-        return { acknowledge: true };
-      })
-      .catch(error => {
-        delete _locks.resetKuzzleData;
-        throw error;
-      });
-  }
-
   /**
    * Reset Redis cache
    */
   resetCache (request) {
-    assertArgsHasAttribute(request, 'database');
-
-    const database = request.input.args.database;
+    const database = this.getStringParam(request, 'args.database');
 
     let cacheEngine;
 
     // @todo allow only publicCache
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)